pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -649,7 +660,10 @@ class Timestamp(_Timestamp):
         else:
             value = self.value
 
-        r = round_ns(value, rounder, freq)
+        value = np.array([value], dtype=np.int64)
+
+        # Will only ever contain 1 element for timestamp
+        r = round_ns(value, rounder, freq).item()
```

Comment:
Nit. I think just indexing into this array (i.e. `round_ns(value, rounder, freq)[0]`) is just fine. Looks like `item` returns a copy of a Python scalar (and we may want to keep this a numpy scalar just in case)

Reply:
`self.value` from `Timestamp` will always be a scalar, so we implicitly know the result of this is be a one element array.